src/main/java/org/jenkinsci/plugins/displayurlapi/DisplayURLContext.java

```diff
@@ -30,6 +35,15 @@
      * The current thread's context.
      */
     private static ThreadLocal<Stack<DisplayURLContext>> context = new ThreadLocal<>();
+
+    private static final Cache<String, Optional<PluginWrapper>> CACHE =
+        CacheBuilder.newBuilder()
+            .maximumSize(Integer.getInteger( DisplayURLContext.class.getName() + ".cache.size", 1000))
+            // cache ttl default 5 minutes
+            .expireAfterAccess( Integer.getInteger( DisplayURLContext.class.getName() + ".cache.ttl", 300000), TimeUnit.MILLISECONDS)
+            //.weakValues() not sure we need this as entries are pluginWrapper will never be garbaged
```

Comment:
Yeah I think you're right, it looks like the lifetime for a `PluginWrapper` is the same as the lifetime of `PluginManager`, so `weakValues` seems pointless.